python/lsst/ip/isr/isrTask.py

```diff
@@ -2127,6 +2133,42 @@ def maskEdges(self, exposure, numEdgePixels=0, maskPlane="SUSPECT"):
                 maskBitMask
             )
 
+    def maskGrowTophat(self, exposure, boxRadius=0, boxThreshold=0.25, maskPlane="BAD", maskLabel="BAD"):
+        """Mask regions around large masked regions.
+
+        Parameters
+        ----------
+        exposure: `lsst.afw.image.Exposure`
+            Exposure to process.
+        boxRadius : `int`, optional
+            Half-size of the box to use to grow masks.
+        boxThreshold : `float`, optional
+            Fraction of box footprint pixels that must be masked on
+            the input to mask on the output.
+        maskPlane : `str`, optional
+            Mask plane name to use for input masking.
+        maskLabel : `str`, optional
+            Mask plan to mark grown regions.
+        """
+        if boxRadius > 0 and boxThreshold > 0.0:
+            maskedImage = exposure.getMaskedImage()
+            maskBitMask = maskedImage.getMask().getPlaneBitMask(maskPlane)
+
+            thresholdImage = afwImage.MaskedImageF(maskedImage.getDimensions())
+            thresholdImage.getImage().getArray()[:, :] = maskedImage.getMask().getArray()[:, :] & maskBitMask
+
+            kernelImage = afwImage.ImageD(2 * boxRadius, 2 * boxRadius)
+            kernelImage.getArray()[:, :] = 1
+
+            convControl = afwMath.ConvolutionControl(False, True, 0)
+            convolvedImage = afwImage.MaskedImageF(maskedImage.getDimensions())
+            afwMath.convolve(convolvedImage, thresholdImage, afwMath.FixedKernel(kernelImage), convControl)
```

Comment:
I've taken the above suggestion to use (2*radius + 1) instead of my initial thought to simply accept the input kernel shape.  Although all the examples I can find have odd dimensions, this guarantees that for any future kernel.